hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java

```diff
@@ -858,9 +859,17 @@ public OzoneKeyDetails getKeyDetails(
     OmKeyInfo keyInfo = ozoneManagerClient.lookupKey(keyArgs);
 
     List<OzoneKeyLocation> ozoneKeyLocations = new ArrayList<>();
-    keyInfo.getLatestVersionLocations().getBlocksLatestVersionOnly().forEach(
-        (a) -> ozoneKeyLocations.add(new OzoneKeyLocation(a.getContainerID(),
-            a.getLocalID(), a.getLength(), a.getOffset())));
+    long lastKeyOffset = 0L;
+    long lastLength = 0L;
+    List<OmKeyLocationInfo> omKeyLocationInfos = keyInfo
+        .getLatestVersionLocations().getBlocksLatestVersionOnly();
+    for (OmKeyLocationInfo info: omKeyLocationInfos) {
+      ozoneKeyLocations.add(new OzoneKeyLocation(info.getContainerID(),
+          info.getLocalID(), info.getLength(), info.getOffset(),
+          lastKeyOffset + lastLength));
+      lastKeyOffset = lastLength + lastKeyOffset;
+      lastLength = info.getLength();
+    }
```

Comment:
Fix me if I am wrong, but based on my understanding the variable usage can be simplified in this way:

```suggestion
    long lastKeyOffset = 0L;
    List<OmKeyLocationInfo> omKeyLocationInfos = keyInfo
        .getLatestVersionLocations().getBlocksLatestVersionOnly();
    for (OmKeyLocationInfo info: omKeyLocationInfos) {
      ozoneKeyLocations.add(new OzoneKeyLocation(info.getContainerID(),
          info.getLocalID(), info.getLength(), info.getOffset(),
          lastKeyOffset));
      lastKeyOffset += info.getLength();
    }
```